### DISCUSSION TITLE
Can't add Azure managed identity

### DISCUSSION THREAD

**Discussion by IanMoroney** [kedacore/charts#260] [Q&A]
When deploying the chart with `--set=podIdentity.activeDirectory.identity=` it won't work, because kubernetes labels have a 63 character limit.
The resultant identity is too long for the label.

## Expected Behavior
If an identity is meant to be used by keda, shouldn't it use its own podIdentity definition?

## Actual Behavior
``
Error: failed to create resource: Deployment.apps "keda-operator" is invalid: [spec.template.labels: Invalid value: "/subscriptions/subscriptionid/resourcegroups/resourcegroupid/providers/Microsoft.ManagedIdentity/userAssignedIdentities/dev-keda": must be no more than 63 characters, spec.template.labels: Invalid value: "/subscriptions/subscriptionid/resourcegroups/resourcegroupid/providers/Microsoft.ManagedIdentity/userAssignedIdentities/dev-keda": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')]
``

**Comment by IanMoroney** [comment]
So the workaround for this is I had to create my own `AzureIdentity` and `AzureIdentityBinding`, deploy it and then refer to it by name when using `--set=podIdentity.activeDirectory.identity=`

Is there a reason why we don't have:
```
--set=podIdentity.activeDirectory.enabled=true
--set=podIdentity.activeDirectory.clientID=<client_id of identity>
--set=podIdentity.activeDirectory.resourceID=<resourceID of identity>
```
instead?

After a lot of experimentation trying to get keda to work, it didn't seem to be documented that an identity was required, but the scaler wouldn't work at all without a managed identity attached.
If it is a requirement, I'd have expected the `podIdentity` configuration to be presented to the helm chart (like with something such as `akv2k8s`, and not to have to define my own podIdentity and binding before deploying the keda chart.

Thoughts?

**Comment by tomkerkhove** [comment]
You do not have to specify the ARM value, instead you need to pass the name of the Kubernetes resource.

If you ant to give this a try, I'd recommend going through this sample: https://github.com/kedacore/sample-dotnet-worker-servicebus-queue/blob/main/pod-identity.md

**Reply by IanMoroney** [reply]
Yes, but my resulting question is why can't we deploy Keda directly with a pod identity and an identity binding? The helm chart could include it so we don't have to deploy an identity as a prerequisite, and it can be turned off by default.

**Reply by tomkerkhove** [reply]
I think that's a great suggestion, but given we are moving towards workload identity in the long-term (see https://github.com/kedacore/keda/issues/2487) I would hold this off until then.

Make sense to you?

**Reply by IanMoroney** [reply]
sure
